### ISSUE TITLE
 ERROR  [nitro] [unhandledRejection] A dynamic import callback was not specified.

### ISSUE THREAD

**Issue by zhangHongEn** [nuxt/nuxt#28911]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v18.20.2
- Nuxt Version:     3.13.1
- CLI Version:      3.13.1
- Nitro Version:    2.9.7
- Package Manager:  pnpm@9.7.0
- Builder:          -
- User Config:      compatibilityDate, debug, devtools, vite
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://github.com/zhangHongEn/nuxt-demo

### Describe the bug
```vue
// app.vue
<template>
  <div>
    <h1>Nuxt host</h1>

    <p>
      Mode1: <i> {{ runtimeConfig.app.buildId ? ' dev ' : ' prod ' }} mode </i>
    </p>
    <hr />
    {{ dd }}
  </div>
</template>

<script setup>
// This line will report an error
new Function("name", "return import(name)")('vm')
const runtimeConfig = useRuntimeConfig();
</script>

```

 ERROR  [nitro] [unhandledRejection] A dynamic import callback was not specified.
 https://github.com/module-federation/core/pull/2934

**Comment by zhangHongEn** [comment]
Dev will report this error, build + preview is normal

**Comment by ScriptedAlchemy** [comment]
You have to pass dynamic import callback or enable a flag in node. 

Check my implementation. If you use older node version you have to add some --experimental flag afik 

The problem is your vm or nested vm is using dynamic import without passing the function to handle it down to vm as well. I got this in node before as well in webpack on newer versions.

**Comment by zhangHongEn** [comment]
@ScriptedAlchemy I think it’s due to certain environments in Nuxt dev mode, because the build works fine.

**Comment by ScriptedAlchemy** [comment]
Yeah i think nuxt utilizes some vm implementation and is missing the flag in our implementation we pass a callback but the current vm context needs to support function imports. Which when running on main threads it does. But in vm you have to pass the callback

**Comment by zhangHongEn** [comment]
@danielroe Can anyone assist with this issue? The reason is that the mf SDK needs to use certain Node modules, and these modules don’t need to be bundled by the packaging tool.
https://github.com/module-federation/core/blob/d5a3072e178d5d12004a0bedf29910a0888f67a3/packages/sdk/src/node.ts#L7

**Comment by Dell-it** [comment]
Good day, everyone. Sorry for the inconvenience, but is there any progress on this issue?  

Recently, I tried integrating `module-federation/vite` with Nuxt 3, and during the build process, it does generate the manifest and `remoteEntry` files. However, their placement does not allow for easy access.  

![Image](https://github.com/user-attachments/assets/66e62172-c329-4879-a695-cd62d4a67523)


But as mentioned earlier, when attempting to run dev, the generation doesn't even occur:
```
[nitro] [unhandledRejection] Only URLs with a scheme in: file, data, and node are supported by the default ESM loader. Received protocol 'virtual:'
```
Thank you for any assistance

**Comment by raminjafary** [comment]
Is there any plan for this to be resolved?

**Comment by danielroe** [comment]
I can no longer reproduce this on main - let me know if you can and I'll happily reopen.
